aztec/src/main/kotlin/org/wordpress/aztec/spans/MarkSpan.kt

```diff
@@ -25,8 +26,9 @@ class MarkSpan : CharacterStyle, IAztecInlineSpan {
         textColorValue = safelyParseColor(colorString)
     }
 
-    private fun safelyParseColor(colorString: String?): Int? {
-        if (colorString == null) {
+    @VisibleForTesting
+    internal fun safelyParseColor(colorString: String?): Int? {
```

Comment:
> I believe this doesn't need to be internal to be testable

I think there is a misunderstanding. I marked it `internal` to decrease the visibility. Went from private to internal. Protected wouldn't work and I need it visible for testing. If you think it being public is fine, I can do that too. Or maybe there is a mockK feature that gets around this annoyance, let me know.

Reply:
What I meant is that even keeping this `private`, which I agree we should opt for the most restrict visibility we can here, you can test this method by calling the constructor of `MarkSpan` that receives a color String as an argument, then check in that created instance if `getTextColor` returns what you expect + the constructor doesn't throw any exceptions.

In other words, treating the class as a black box and test behaviors, inputs, and outputs, instead of making a specific internal method visible for testing and white-box testing its source code.

---

Besides that, I think I anticipated myself a bit in my first comment mentioning `super()`, but what I meant is that I see that this class is a subclass and I didn't check the constructor of the superclass to affirm that calling the constructor of `MarkSpan` from a test class can cause any issues, in which case it could make sense to bump the visibility to `internal` so we can at least test something.

Reply:
Great point. And I think we may be getting into the weeds here. What I did was on purpose and I specifically meant to test just a method. 

When I approach unit tests I try and test a very specific unit of code. What if the implementation of one of the constructors changes? For example adding a new required parameter? The test could break making it brittle. If we test a specific method, it's independent of any business logic changes since we are testing just a function. And more importantly, we really only care about the color parsing which is what this crash is about.

Additionally we would no longer have the ability to check the return type of `safelyParseColor`. We'd need to in the end expose `textColorValue`, which in essence is equivalent to exposing the method `safelyParseColor`. 

With all that said, it sounds like you feel strongly about running the test a specific way so I made the changes that I think you wanted. I've never used MockK, so let me know if there are any further changes.

Reply:
> When I approach unit tests I try and test a very specific unit of code. What if the implementation of one of the constructors changes? For example adding a new required parameter? The test could break making it brittle. If we test a specific method, it's independent of any business logic changes since we are testing just a function. And more importantly, we really only care about the color parsing which is what this crash is about.

This is a valid point as well! 

I tend to approach test with the class usage in mind, treating everything private as implementation details, which are subject to change at any time. Or as many say: "test behavior, not implementation". The testable "units" for me are any public APIs an external code can use to interact with the class.

For instance, in this situation, in the app perspective, the crash happened when instantiating `MarkSpan` (which in turn calls private/internal methods to initialize and set up itself). The calling code only interacts with the constructor and with the `getTextColor` public method, so it makes sense to call and assert using only them.

But what you said is totally valid as well, one might be more interested that each function inside a class work as expected, especially if they are more like scoped utility functions that are used several places inside that class. And that works fine especially in lib code (like this) and modularized projects, where `internal` helps with keeping things private to external callers.

Anyway, I think this was a nice discussion and I personally think the code looks cleaner now in this last iteration, so I will just reiterate my approval for this last code review.

Thanks for the changes! 🙇🏼